spec/src/main/asciidoc/WebServicesMetadata.adoc

```diff
@@ -943,2259 +807,1399 @@ header, and the mode is OUT.
-<s:complexType name="PingDocument">
+It is an error to combine this annotation with the @SOAPMessageHandlers
+annotation.
 
-. . .
+The @HandlerChain annotation MAY be present on the endpoint interface
+and service implementation bean. The service implementation bean’s @HandlerChain is
+used if @HandlerChain is present on both.
 
-</s:complexType>
+The @HandlerChain annotation MAY be specified on the type only. The
+annotation target includes METHOD and FIELD for use by Jakarta XML Web Services 2.3 [5]. A
+Jakarta Web Services Metadata Processor is REQUIRED to report an error if the @HanderChain
+annotation is used on a method.
 
-<s:complexType name="SecurityHeader">
+The @HandlerChain annotation contains the following member-values:
 
-. . .
+[cols=3, options=header]
+|===
+|Member-Value
+|Meaning
+|Default
 
-</s:complexType>
+|File
+|Location of the handler chain file. The location supports 2 formats.
 
-<s:element name="SecurityHeader" type="SecurityHeader"/>
+1. An absolute java.net.URL in externalForm.
+(ex: http://myhandlers.foo.com/handlerfile1.xml)
 
-</s:schema>
+2. A relative path from the source file or class file. (ex:
+bar/handlerfile1.xml)
+|None
 
-</ types>
+|name
+|*Deprecated* as of Jakarta Web Services Metadata 2.0 with no replacement.
 
-<message name="PingOneWay">
+The name was originally used to associate a Jakarta XML RPC handler in a handler
+chain with the web service it is declared in. Jakarta XML Web Services handlers are
+associated to Web Services through elements in the handler chain itself.
+In this version, the name is ALWAYS ignored.
 
-<part name="arg0" type="tns:PingDocument"/>
+This member-value will be permanently removed in a future version of
+Jakarta Web Services Metadata.
 
-</message>
+|””
+|===
 
-<message name="PingTwoWay">
+==== Annotation Type Definition
 
-<part name="arg0" type="tns:PingDocument"/>
+[source,java]
+----
+@Retention(value=RetentionPolicy.RUNTIME)
+@Target({TYPE, METHOD, FIELD})
+public @interface HandlerChain {
+  String file();
+  String name() default "";
+};
+----
 
-</message>
+==== Examples
 
-<message name="PingTwoWayResponse">
+Example 1
 
-<part name="arg0" type="tns:PingDocument"/>
+*Java Source:*
 
-</message>
+Located in /home/mywork/src/com/jwsm/examples/
 
-<message name="SecurePing">
+[source,java]
+----
+package com.jwsm.examples
 
-<part name="arg0" type="tns:PingDocument"/> <part name="arg1"
-element="tns:SecurityHeader"/> </message>
+@WebService
+@HandlerChain(file="config/ProjectHandlers.xml")
+public class MyWebService {
+};
+----
 
-<portType name="PingService">
+*Handler Chain Configuration File*
 
-<operation name="PingOneWay">
+Located in /home/mywork/src/com/jwsm/examples/config/
 
-<input message="tns:PingOneWay"/> +
-</operation>
+[source,xml]
+----
+<?xml version="1.0" encoding="UTF-8" standalone="yes"?> 
+<bindings
+  wsdlLocation="http://localhost:8080/jaxrpc-fromwsdl_handler/test?wsdl"
```

Comment:
avoid using "jaxrpc", let the URL be just "http://localhost:8080/fromwsdl_handler/test?wsdl"

Reply:
ok, resolved by https://github.com/eclipse-ee4j/jws-api/pull/69/commits/62fdf397f28a148826367da36bcaef2c30e97c7d